torch/csrc/autograd/engine.cpp

```diff
@@ -152,15 +152,21 @@ struct ReadyQueue {
 // the historic behavior.
 
 int NodeTask::getReentrantDepth() const {
-  return base_->reentrant_depth_;
+  std::shared_ptr<GraphTask> graph_task;
+  TORCH_INTERNAL_ASSERT(
+      graph_task = base_.lock(), "GraphTask is no longer valid!")
+  return graph_task->reentrant_depth_;
 }
 
 auto ReadyQueue::push(NodeTask item, bool incrementOutstandingTasks) -> void {
   {
     // Lock mutex for writing to heap_
     std::lock_guard<std::mutex> lock(mutex_);
+    std::shared_ptr<GraphTask> graph_task;
+    TORCH_INTERNAL_ASSERT(
+        graph_task = item.base_.lock(), "GraphTask is no longer valid!")
```

Comment:
Hot take: `push()` should take `GraphTask*` as an argument. If I'm pushing a task onto the ready queue, the graph task ought to be live.